python/ray/util/sgd/torch/distributed_torch_runner.py

```diff
@@ -220,28 +228,23 @@ def reserve_cuda_device(retries=20):
         assert isinstance(cuda_devices, str)
         cuda_device_set = set(cuda_devices.split(","))
 
-    global _dummy_actor
-    unused_actors = []
-
     success = False
     for _ in range(retries):
         if _dummy_actor is None:
             _dummy_actor = ray.remote(
-                num_gpus=1,
+                num_cpus=num_cpus, num_gpus=1,
                 resources={"node:" + ip: 0.1})(_DummyActor).remote()
 
-        reserved_device = ray.get(_dummy_actor.cuda_devices.remote())
+        reserved_gpu_device = ray.get(_dummy_actor.cuda_devices.remote())
 
-        if match_devices and reserved_device not in cuda_device_set:
+        if match_devices and reserved_gpu_device not in cuda_device_set:
             logger.debug("Device %s not in list of visible devices %s",
-                         reserved_device, cuda_device_set)
-            unused_actors.append(_dummy_actor)
+                         reserved_gpu_device, cuda_device_set)
+            _dummy_actor.__ray_terminate__.remote()
             _dummy_actor = None
         else:
-            logger.debug("Found matching device %s", reserved_device)
+            logger.debug("Found matching device %s", reserved_gpu_device)
             success = True
```

Comment:
can you please revert this change?

Reply:
Now the dummy actor reserve both CPU and GPU resources. The dummy actors will be reserved many of CPU resources if GPU reserving failed multiple times. Does this expect?

Reply:
reverted